core/chains/evm/client/node_lifecycle.go

```diff
@@ -305,17 +320,21 @@ func (n *node) unreachableLoop() {
 		case <-n.chStop:
 			return
 		case <-time.After(dialRetryBackoff.Duration()):
-			lggr.Tracew("Trying to re-dial RPC node", "nodeState", n.State())
+			lggr.Tracew("Trying to re-dial RPC node", "nodeState", n.getState())
 
-			err := n.dial(context.Background())
+			ctx, cancel := utils.ContextFromChan(n.chStop)
```

Comment:
I wonder if there could be a global context on the node that is canceled on stop, that might mean we can avoid spinning a new goroutine every time and being bery careful with calling `cancel`?

Reply:
Done.